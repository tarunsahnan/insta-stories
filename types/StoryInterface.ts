export interface StoryInterface {
  userName: string;
  userImage: string;
  userStory: string;
}
