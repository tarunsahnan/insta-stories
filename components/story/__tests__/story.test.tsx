import { describe, expect, test, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import Story from "../story";
import { StoryInterface } from "@/types/StoryInterface";

const mockStory: StoryInterface = {
  userName: "John Doe",
  userImage: "https://example.com/image.jpg",
  userStory: "https://example.com/story.jpg",
};

describe("Story component", () => {
  test("renders story with correct username and image", () => {
    const { container, rerender } = render(
      <Story item={mockStory} onClickHandler={() => {}} />
    );
    expect(container).toMatchSnapshot();
    const usernameElement = screen.getByText(mockStory.userName);
    const imageElement = screen.getByAltText(mockStory.userName);
    expect(usernameElement).toBeDefined();
    expect(imageElement).toBeDefined();
    expect(imageElement.attributes);
  });

  test("triggers onClickHandler when clicked", () => {
    const onClickHandlerMock = vi.fn();
    render(<Story item={mockStory} onClickHandler={onClickHandlerMock} />);

    const storyElement = screen.getAllByTestId("story");
    fireEvent.click(storyElement[1]);
    expect(onClickHandlerMock).toHaveBeenCalled();
  });
});
