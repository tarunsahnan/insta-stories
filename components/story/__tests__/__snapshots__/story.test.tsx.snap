// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Story component > renders story with correct username and image 1`] = `
<div>
  <div
    data-testid="story"
  >
    <div
      class="bg-gradient-to-tr from-yellow-500 to-red-600 p-[1.5px]  rounded-full"
    >
      <div
        class="bg-white rounded-full p-1"
      >
        <img
          alt="John Doe"
          class="w-14 h-14 rounded-full object-cover"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=128&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=128&q=75 2x"
          style="color: transparent;"
          width="50"
        />
      </div>
    </div>
    <p
      class="text-xs w-16 truncate text-center"
    >
      John Doe
    </p>
  </div>
</div>
`;
